cpp/CMakeLists.txt

```diff
@@ -157,7 +157,7 @@ add_library(cudf
     src/bitmask/null_mask.cu
     src/column/column.cu
     src/column/column_device_view.cu
-    src/column/column_factories.cpp
+    src/column/column_factories.cu
```

Comment:
I believe the current recommendation is to split `column_factories` into two separate files so that only the minimal subset of functions that require `CUDA` are in the .cu